.github/workflows/docker-build-and-run.yml

```diff
@@ -0,0 +1,33 @@
+name: Build and Run Docker
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-and-run-docker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      # Build the Docker image
+      - name: Build Docker image
+        run: |
+          docker build -t tf2.4_ivim-mri_codecollection -f Docker/Dockerfile .
+
+      # Run the Docker container
+      - name: Run Docker container
+        run: |
+          docker run -it --rm --name TF2.4_IVIM-MRI_CodeCollection \
```

Comment:
The check failed, it's saying the terminal is not a TTY, so I think drop the `-it` or just the `t` part of that?